UnitedIncome/components
# [APP-2933] BoxSelectCard

### Description
> Please provide a brief description of your changes below.

Implements the `BoxSelectCard` which lets you make a box selection within a question card.

![gofff22222](https://user-images.githubusercontent.com/10888441/60212961-817fcf80-9830-11e9-8b78-66c1dba75800.gif)

### Testing Instructions
> What should be done in order to test this pull request, or re-create its effects.

* Load up the card component in storybook.
* Test it in a flow.

### Comments
> Any additional comments that will aid with the review process.

Also adds a focus state to the Box component. To handle this I've had to make it stateful.

- [JIRA Ticket](https://unitedincome.atlassian.net/browse/app-2933)
